### DISCUSSION TITLE
EPG Fetch Quits / Exits Without Error or Warning

### DISCUSSION THREAD

**Discussion by nomaam** [iptv-org/community#1026] [Q&A]
I have tried running the following command through cron and watching it execute through a command prompt, and for some reason it just quits without error

    npm run grab -- --channels=../gen/channels.xml --maxConnections=10 --output ../gen/epg.xml

I have tried absolute and relative paths without any change.

    config:
    output: ../gen/epg.xml
    maxConnections: 10
    gzip: false
    channels: ../gen/channels.xml
    loading channels...
    found 10933 channels
    creating queue...
    added 22459 items
    run #1:
    (node:19357) Warning: Setting the NODE_TLS_REJECT_UNAUTHORIZED environment variable to '0' makes TLS connections and HTTPS requests insecure by disabling certificate verification.
    (Use `node --trace-warnings ...` to show where the warning was created)
    [1/22459]

Then at around 5500/22459 the script just exits. sometimes it gets a bit higher in the count or lower, different each time. No error or warning shown. The script also doesn't generate any output file with the epg data. 

Thoughts? I am at a loss as to what the issues could be. I am running on Debian. 

Thanks

**Comment by freearhey** [comment]
Most likely it's a matter of exceeding the limit of RAM available to Node.js.

One way to get around this limitation is to increase the limit via an environment variable. Example:

```
NODE_OPTIONS=--max-old-space-size=8192 npm run grab -- --channels=../gen/channels.xml --maxConnections=10 --output ../gen/epg.xml
```

(8192 = 8Gb)

Another way is to use a custom list with only the desired channels:

https://github.com/iptv-org/epg?tab=readme-ov-file#use-custom-channel-list
